Dockerfile

```diff
@@ -2,15 +2,31 @@
 # Use of this source code is governed by a BSD-style
 # license that can be found in the LICENSE file.
 
-FROM golang:1.13.5-alpine AS builder
-
+FROM golang:1.13.5 AS tester
 WORKDIR /code
-ENV CGO_ENABLED=0
 ADD go.mod go.sum ./
 RUN go mod download
 ADD . ./
+RUN go test -race -vet all -mod readonly ./...
+
+FROM golangci/golangci-lint AS linter
+WORKDIR /code
+COPY --from=tester /go/pkg /go/pkg
+COPY --from=tester /code .
+RUN golangci-lint run --enable-all \
+    -D errcheck -D lll -D dupl -D gochecknoglobals -D unparam \
+    --deadline 5m \
```

Comment:
This has been a recent addition: i realized i was consuming a lot of space with docker images, and i realized this linter layer was storing more than 200M on each docker build after any code change... it was downloading all go packages again, and also golangci stores a cache to speed up future executions we will never reuse inside a docker layer. This has also improved performance

Reply:
Can you make it just ``golangci-lint run``? That way it uses the config specified in golangci.yml.

Reply:
Sorry, I didnt realize golangci-lint config file was introduced in master after my original branch (side effects of having this in my plate for so long :( ). I see you amended in master: I will start rebasing the subsequent PRs... i guess its going to take my entire life to resolve the conflicts :) News soon.

Reply:
You can `rebase -i` and drop the commits that were merged in this PR :D

Reply:
Thanks for the suggestion. Yeah... but what worries me are the commits post-review, which are not in the other branches yet, and given the incremental nature, probably are going to touch lines that other PRs are also modifying. In any case: the other PRs are quite small + (business as usual && life is hard), nothing to worry :)